src/sql/src/plan/statement/ddl.rs

```diff
@@ -2176,10 +2176,13 @@ pub fn plan_create_type(
         CreateTypeAs::Record { ref column_defs } => {
             for column_def in column_defs {
                 let key = ident(column_def.name.clone());
-                let (data_type, dt_ids) =
-                    resolve_names_data_type(scx, column_def.data_type.clone())?;
-                ensure_valid_data_type(scx, data_type, &as_type, &key)?;
-                ids.extend(dt_ids);
+                let (data_type, _) = resolve_names_data_type(scx, column_def.data_type.clone())?;
+                ensure_valid_data_type(scx, &data_type, &as_type, &key)?;
+                if let ResolvedDataType::Named { id, .. } = data_type {
+                    ids.push(id);
+                } else {
```

Comment:
I believe this is correct as written, but in the interest of future proofing and clarifying intent, you should split `ids` into two separate vectors. The thing currently called `ids` should be called `depends_on`, and you should `depends_on.extend(dt_ids)`. Then you should introduce a _new_ field called `field_ids`, and push the `id` you have here into that. 

Does that make sense? The idea is that `resolve_names_data_type` returns the set of all observed IDs. You're not meant to do anything with these IDs except aggregate them and record them in the `depends_on` field. Metadata about the resolved type should be fetched via `data_type`, exactly as you're now doing.

Also, yes, this is the most 🦶🏽🔫  API imaginable. We got 80% of the way through splitting name resolution out into a separate pass, and not the last 20% of the way. (See #5302.) Someone ought to clean this all up at some point, but for now we just deal with it. 🤷🏽‍♂️ Specifically, the `resolve_names` call should happen once at the VERY beginning of planning, the dependent IDs recorded, and then the rest of planning should not have to worry about handling name resolution or dependent IDs at all!

Reply:
Oh yeah that's way better, will update